src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -189,6 +222,34 @@ public static RequestDelegateResult Create(MethodInfo methodInfo, Func<HttpConte
         return HandleRequestBodyAndCompileRequestDelegate(responseWritingMethodCall, factoryContext);
     }
 
+    private static Func<RouteHandlerFilterContext, ValueTask<object?>> CreateFilterPipeline(MethodInfo methodInfo, Expression? target, FactoryContext factoryContext)
+    {
+        // httpContext.Response.StatusCode == 400
+        // ? Task.CompletedTask
+        // : handler((string)context.Parameters[0], (int)context.Parameters[1])
+        var filteredInvocation = Expression.Lambda<Func<RouteHandlerFilterContext, ValueTask<object?>>>(
+            Expression.Condition(
+                Expression.Equal(FilterContextHttpContextStatusCodeExpr, Expression.Constant(400)),
```

Comment:
We're going to make [team 422](https://httpstatuses.com/422) sad.

Reply:
I wonder if this should be >= 400

Reply:
> We're going to make [team 422](https://httpstatuses.com/422) sad.

The current logic already assumes `400` so no sadder than usual.

> I wonder if this should be >= 400

Interesting. It makes more sense to do this specifically for the case where filters have run because there is a chance the filter might set the status code to 500 or greater to indicate a failure in the filter itself.

Reply:
+1 for `>= 400` being better than `== 400`. We'll hopefully be using something other than the status code for this by the final .NET 7 release though.

Reply:
Update the conditional.